engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```diff
@@ -769,7 +769,11 @@ public void processHeartbeatResponse(ContainerHeartbeatResponse rsp)
     }
 
     if (rsp.committedWindowId != lastCommittedWindowId) {
+
       lastCommittedWindowId = rsp.committedWindowId;
+
+      bufferServer.purge(lastCommittedWindowId);
```

Comment:
While currently buffer server by itself does not make any assumption regarding locality and communicates with its clients (publisher, subscriber, controller) over network request API, Apex supports deployment of the buffer server into the same container as the upstream operator/publisher only. As the buffer server is not fault tolerant, and there is no plan to make it fault tolerant, in reality, there is no flexibility to deploy buffer server into another container other than the upstream operator container. Additionally, as long as the buffer server continues to support purge request submitted over a network, the flexibility to deploy buffer server in any container still can be exercised once (or if) fault tolerance is supported by the buffer server.